llvm/test/CodeGen/RISCV/rvv/vsetvli-insert.ll

```diff
@@ -329,9 +329,9 @@ entry:
 define double @test17(i64 %avl, <vscale x 1 x double> %a, <vscale x 1 x double> %b) nounwind {
 ; CHECK-LABEL: test17:
 ; CHECK:       # %bb.0: # %entry
-; CHECK-NEXT:    vsetvli a0, a0, e64, m1, ta, ma
-; CHECK-NEXT:    vfmv.f.s fa5, v8
+; CHECK-NEXT:    vsetvli a0, a0, e32, mf2, ta, ma
```

Comment:
This can be removed too? It just sets `vl`, `vtype` will be changed by `vsetvli` below.

Reply:
This was due to a use by an instruction which was marked for deletion. Unfortunately, to account for such cases I needed to make `ToDelete` a `SmallPtrSet` so that I could quickly discard instructions added to this set: https://github.com/llvm/llvm-project/pull/67144/commits/e8f2fbe5904f9fb58a75e8ca7c5e82c4a11f3c1c